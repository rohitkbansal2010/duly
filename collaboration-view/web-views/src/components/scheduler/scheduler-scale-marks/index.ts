export { SchedulerScaleMarks } from './scheduler-scale-marks';
export type { SchedulerScaleMarksPropsType } from './scheduler-scale-marks';
