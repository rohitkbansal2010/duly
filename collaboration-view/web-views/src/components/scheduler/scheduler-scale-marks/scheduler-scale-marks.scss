@import "~@styles/colors.scss";
@import "~@styles/typography.scss";

.schedulerScaleMarks {
  &Item {
    height: 15.625rem;
    padding: 0.25rem 0 0.5rem 0 !important;
    background: none !important;
    border: 0 !important;

    &:last-child {
      height: 0;
    }

    & div {
      display: flex !important;
      align-items: center !important;
    }

    &Division {
      flex: 1 0 0%;
      width: 100%;
      max-width: 6.25rem;
      color: $medium-gray;
      font-size: 1rem;
      line-height: 1.5rem;
      @include nexa-extra-bold-highlight;
    }

    &Line {
      z-index: 15;
      flex: 1 0 0%;
      width: 100%;
      border-top: 0.13rem dashed $medium-gray !important;
    }
  }
}
