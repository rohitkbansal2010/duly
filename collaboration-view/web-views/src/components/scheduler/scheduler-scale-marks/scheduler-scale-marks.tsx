import { ListGroup, ListGroupItem } from 'react-bootstrap';

import { calculateCalendarDivisions } from '@utils';

import styles from './scheduler-scale-marks.scss';

export type SchedulerScaleMarksPropsType = {
  workDayStart: number;
  workDayEnd: number;
};

export const SchedulerScaleMarks = ({
  workDayStart,
  workDayEnd,
}: SchedulerScaleMarksPropsType) => {
  const schedulerDivisions = calculateCalendarDivisions(workDayStart, workDayEnd);

  const renderListItem = (division: string) =>
    (
      <ListGroupItem key={division} className={styles.schedulerScaleMarksItem}>
        <div>
          <div className={styles.schedulerScaleMarksItemDivision}>
            {division}
          </div>
          <div className={styles.schedulerScaleMarksItemLine} />
        </div>
      </ListGroupItem>
    );

  return (
    <ListGroup variant="flush" className={styles.schedulerScaleMarks}>
      {schedulerDivisions.map(renderListItem)}
    </ListGroup>
  );
};
