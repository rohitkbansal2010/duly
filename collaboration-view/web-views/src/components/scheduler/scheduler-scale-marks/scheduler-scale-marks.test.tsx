import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { calculateCalendarDivisions } from '@utils';

import { SchedulerScaleMarks, SchedulerScaleMarksPropsType } from './scheduler-scale-marks';

const mockProps: SchedulerScaleMarksPropsType = {
  workDayStart: 7,
  workDayEnd: 19,
};

describe('SchedulerScaleMarks', () => {
  let component: ReturnType<typeof shallow>;

  beforeEach(() => {
    component = shallow(<SchedulerScaleMarks {...mockProps} />);
  });

  it('should render correctly', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render the "ListGroup" component from "react-bootstrap"', () => {
    expect(component.find('ListGroup')).toHaveLength(1);
  });

  it('should render time segments to the "ListGroup" component', () => {
    const listGroup = component.find('ListGroup');
    const schedulerDivisions =
      calculateCalendarDivisions(mockProps.workDayStart, mockProps.workDayEnd);

    expect(listGroup.find('ListGroupItem')).toHaveLength(schedulerDivisions.length);
  });
});
