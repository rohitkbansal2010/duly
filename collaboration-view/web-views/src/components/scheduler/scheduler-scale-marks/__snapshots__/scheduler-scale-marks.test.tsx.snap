// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchedulerScaleMarks should render correctly 1`] = `
<ListGroup
  className="schedulerScaleMarks"
  variant="flush"
>
  <ListGroupItem
    className="schedulerScaleMarksItem"
    key="7:00 am"
  >
    <div>
      <div
        className="schedulerScaleMarksItemDivision"
      >
        7:00 am
      </div>
      <div
        className="schedulerScaleMarksItemLine"
      />
    </div>
  </ListGroupItem>
  <ListGroupItem
    className="schedulerScaleMarksItem"
    key="8:00 am"
  >
    <div>
      <div
        className="schedulerScaleMarksItemDivision"
      >
        8:00 am
      </div>
      <div
        className="schedulerScaleMarksItemLine"
      />
    </div>
  </ListGroupItem>
  <ListGroupItem
    className="schedulerScaleMarksItem"
    key="9:00 am"
  >
    <div>
      <div
        className="schedulerScaleMarksItemDivision"
      >
        9:00 am
      </div>
      <div
        className="schedulerScaleMarksItemLine"
      />
    </div>
  </ListGroupItem>
  <ListGroupItem
    className="schedulerScaleMarksItem"
    key="10:00 am"
  >
    <div>
      <div
        className="schedulerScaleMarksItemDivision"
      >
        10:00 am
      </div>
      <div
        className="schedulerScaleMarksItemLine"
      />
    </div>
  </ListGroupItem>
  <ListGroupItem
    className="schedulerScaleMarksItem"
    key="11:00 am"
  >
    <div>
      <div
        className="schedulerScaleMarksItemDivision"
      >
        11:00 am
      </div>
      <div
        className="schedulerScaleMarksItemLine"
      />
    </div>
  </ListGroupItem>
  <ListGroupItem
    className="schedulerScaleMarksItem"
    key="12:00 pm"
  >
    <div>
      <div
        className="schedulerScaleMarksItemDivision"
      >
        12:00 pm
      </div>
      <div
        className="schedulerScaleMarksItemLine"
      />
    </div>
  </ListGroupItem>
  <ListGroupItem
    className="schedulerScaleMarksItem"
    key="1:00 pm"
  >
    <div>
      <div
        className="schedulerScaleMarksItemDivision"
      >
        1:00 pm
      </div>
      <div
        className="schedulerScaleMarksItemLine"
      />
    </div>
  </ListGroupItem>
  <ListGroupItem
    className="schedulerScaleMarksItem"
    key="2:00 pm"
  >
    <div>
      <div
        className="schedulerScaleMarksItemDivision"
      >
        2:00 pm
      </div>
      <div
        className="schedulerScaleMarksItemLine"
      />
    </div>
  </ListGroupItem>
  <ListGroupItem
    className="schedulerScaleMarksItem"
    key="3:00 pm"
  >
    <div>
      <div
        className="schedulerScaleMarksItemDivision"
      >
        3:00 pm
      </div>
      <div
        className="schedulerScaleMarksItemLine"
      />
    </div>
  </ListGroupItem>
  <ListGroupItem
    className="schedulerScaleMarksItem"
    key="4:00 pm"
  >
    <div>
      <div
        className="schedulerScaleMarksItemDivision"
      >
        4:00 pm
      </div>
      <div
        className="schedulerScaleMarksItemLine"
      />
    </div>
  </ListGroupItem>
  <ListGroupItem
    className="schedulerScaleMarksItem"
    key="5:00 pm"
  >
    <div>
      <div
        className="schedulerScaleMarksItemDivision"
      >
        5:00 pm
      </div>
      <div
        className="schedulerScaleMarksItemLine"
      />
    </div>
  </ListGroupItem>
  <ListGroupItem
    className="schedulerScaleMarksItem"
    key="6:00 pm"
  >
    <div>
      <div
        className="schedulerScaleMarksItemDivision"
      >
        6:00 pm
      </div>
      <div
        className="schedulerScaleMarksItemLine"
      />
    </div>
  </ListGroupItem>
  <ListGroupItem
    className="schedulerScaleMarksItem"
    key="7:00 pm"
  >
    <div>
      <div
        className="schedulerScaleMarksItemDivision"
      >
        7:00 pm
      </div>
      <div
        className="schedulerScaleMarksItemLine"
      />
    </div>
  </ListGroupItem>
</ListGroup>
`;
